documentation/general/decouple-vs-and-net-sdk.md

```diff
@@ -0,0 +1,180 @@
+
+Design Goals:
+
+1. **Consistent Build Experiences**: The `msbuild` and `dotnet build` command line build experience for a solution should be functionally equivalent.
+2. **Independent Design Time Behavior**: Visual Studio's design-time experience should be independent of the .NET SDK used by the solution.
+3. **Clear Handling of Divergence**: To make explicit that it is okay, and even expected, that the design time and command line build experiences can differ when the two products are not in sync.
+
+## Terminology
+
+- **msbuild**: refers to the .NET Framework-based msbuild included with Visual Studio.
+- **dotnet build**: refers to the .NET Core based msbuild included with the .NET SDK.
+- **analyzers**: this document will use analyzers to refer to both analyzers and generators.
+- **torn state**: when the .NET SDK and Visual Studio are not in sync.
+- **.NET SDK project**: a project that uses the .NET SDK project format.
+- **anaylzers**: refer to both analyzers and generators unless otherwise specified.
+
+## Summary
+
+Visual Studio and the .NET SDK are separate products but they are intertwined in command line and design time build scenarios as different components are loaded from each product. This table represents how intertwined Roslyn specifically is currently:
+
+| Scenario | Loads Roslyn | Loads Analyzers / Generators |
+| --- | --- | --- |
+| msbuild | From Visual Studio | From .NET SDK |
+| dotnet build | From .NET SDK | From .NET SDK |
+| Visual Studio Design Time | From Visual Studio | From Both |
+
+There are other products that have similar intertwined behavior such as NuGet, MSBuild, Razor, etc ... Roslyn is generally the most impactful and the one we are focusing on in this document as its path forward can be generalized to the other products.
+
+Generally this mixing of components is fine because Visual Studio will install a .NET SDK that is functionally paired with it. For example 17.10 installs .NET SDK 8.0.3xx, 17.9 installs .NET SDK 8.0.2xx, etc ... In that scenario the components are the same version and the mixing is largely benign. However when the customer is in a torn state, that is the .NET SDK used is not paired with the Visual Studio version, then compatibility issues can, and **will**, arise. Particularly when the .NET SDK is _newer_ than Visual Studio customers will end up with the following style of error:
+
+> CSC : warning CS9057: The analyzer assembly '..\dotnet\sdk\8.0.200\Sdks\Microsoft.NET.Sdk.Razor\source-
```

Comment:
Technically, this should not happen anymore since we have the framework toolset package. Should this document mention that package btw?

Reply:
Added a section about this.